src/oop/object_independence/medium/q2/MainTest.java

```diff
@@ -0,0 +1,172 @@
+package oop.object_independence.medium.q2;
+
+import global.ObjectTest;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class MainTest {
+    // Java
```

Comment:
This says its a Java question, but the question description says it's a parsons. Since it's a medium question, I imagine it's supposed to be parsons

Reply:
Yes because we decided to make all the medium object questions to java with partial code, and parsons only be part of the hard questions